torch/optim/lr_scheduler.py

```diff
@@ -1,23 +1,40 @@
 import types
 import math
-from torch._six import inf
-from functools import partial, wraps
 import warnings
+from collections import Counter
+from functools import partial, wraps
 from bisect import bisect_right
+from torch._six import inf
 
 from .optimizer import Optimizer
 
 
+EPOCH_DEPRECATION_WARNING = ("The epoch parameter is being deprecated. If epoch "
+                             "is different from None, the closed form is used "
+                             "whenever available instead of the chainable form.")
+
+
+def _attach_opt(self, optimizer):
+    if not isinstance(optimizer, Optimizer):
+        raise TypeError('{} is not an Optimizer'.format(
+            type(optimizer).__name__))
+    self.optimizer = optimizer
+
+
 class _LRScheduler(object):
+
     def __init__(self, optimizer, last_epoch=-1):
-        if not isinstance(optimizer, Optimizer):
-            raise TypeError('{} is not an Optimizer'.format(
-                type(optimizer).__name__))
-        self.optimizer = optimizer
+        _attach_opt(self, optimizer)
+        self._init_epoch_and_base(optimizer, last_epoch)
+        self._ensure_opt_first_helper()
+        self.step()
+
+    def _init_epoch_and_base(self, optimizer, last_epoch):
         if last_epoch == -1:
             for group in optimizer.param_groups:
                 group.setdefault('initial_lr', group['lr'])
-            last_epoch = 0
+            # last_epoch = 0
+            # step_last_epoch = None
```

Comment:
Kill these lines? (Unless you have a reason to keep them?)

Reply:
Thanks for pointing this out: I'll remove them.